### PR TITLE
Add CI for build refactor

### LINKED ISSUE #94: Add Travis CI
:construction_worker: Add automated build of docker image with Travis CI

### PR DESCRIPTION
**Pull request by madmath03**
:construction: **Still in progress**

Objective is to close #94 by adding to CI:
- [x] release build
- [x] release run
- [x] unit tests placeholders

### CODE REVIEW

**Comment by madmath03** [comment]
Hi @niwinz,

For information, I'm currently building on the side the docker repositories we discussed in #97:
* https://github.com/Monogramm/docker-uxbox-frontend
* https://github.com/Monogramm/docker-uxbox-backend

Frontend is working fine, but backend is giving me a bit more trouble so far, even though everything seems to work fine here...
Anyway, I will make it work, it's just a matter of time.

What I was wondering though was about the change on how the backend is delivered. I'm really not used to Clojure so I was wondering : what's the benefit of deploying the backend as source instead of packaging the backend as an uberjar ?
The latest seemed a little faster / safer to me since all the libraries were already in archive, but, again, there is a lot I do not know about Clojure 😅 

In the meantime, I should be able to finish this PR with the unit tests soon.

**Comment by niwinz** [comment]
hi @madmath03 
The reason to deploy as source instead of uberjar, is because some dependencies (bouncycastle) does not works properly under "fatjar" (bacause they are cryptographically signed, and when it is repackaged, the signature is no longer valid). Additionally, deploying as source will allow in future things like deploy with 0 downtime or patch live code (in the same way as erlang can do it). If this were proprietary project deployed on third party, i can understand the no show the source, but this is an open source project, so this is not a problem. 

On the end deploying a source is very similar to previous jar file, with exception that is not packaged into single jar.

**Comment by niwinz** [comment]
@madmath03 if you need some help, just tell me, maybe we can join some chat, for to have more direct contact?

**Comment by niwinz** [comment]
@madmath03 i think that the best direction is leave in this PR the unit test part as "placeholders" and let me finish the `back/api-refactor` PR because i'm changing a lot of stuff there and tests are affected.

**Comment by madmath03** [comment]
@niwinz regarding adding Unit Tests as "placeholders", that's also what I was thinking and started doing in the latest commit of this PR.

I wanted to have a try at calling the existing tests by myself, mostly for training / learning more about Clojure UT, but I might not have the time for that, and as you said, probably better to leave it to you.

I will check if everything is clean in this PR and send it to you.

Also, regarding your offer for chat, thanks a lot. I might send you an email if I'm stuck at some point 😉

**Comment by madmath03** [comment]
Added a script in backend to prepare the release. The idea is to keep the inteeligence out of the manage.sh script and delegate to another script as much as possible. We can then use this script to more easily build a Docker container.

@niwinz All good for me here. I let review this and merge it if that's fine for you.

File: manage.sh [hunk 1]
```diff
@@ -46,14 +49,74 @@ function run-devenv {
...
+
 function build-release-frontend-local {
+    if ! $(docker images | grep $IMGNAME | grep -q $REV); then
+        build-devenv
+    fi
+
+    mkdir -p $HOME/.m2
+    rm -rf ./frontend/node_modules
+    # FIXME Ugly... should be a better way
+    chmod 777 ./frontend
```
Comment (niwinz) [inline manage.sh:107]: why this `chmod 777` ? do you have permissions related problem with that mounted directories?

Consider building the image with `--build-arg EXTERNAL_UID=$(id -u) ` (i will add it to the default build-devenv command. This parameter will create the image user with the same id as in your machine, so your permission problems will disappear.

Comment (madmath03) [reply manage.sh:107]: Indeed I've had permissions issues when building the frontend (don't remember which files exactly though) so I added the FIXME as a reminder.

Your idea should probably do the trick. I will try it a bit later this week and let you know.

**Pull request merged**